{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# If not installed, run this cell. It is needed for access to GEE\n",
    "!pip install ee\n",
    "!pip show ee"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "            <style>\n",
       "                .geemap-dark {\n",
       "                    --jp-widgets-color: white;\n",
       "                    --jp-widgets-label-color: white;\n",
       "                    --jp-ui-font-color1: white;\n",
       "                    --jp-layout-color2: #454545;\n",
       "                    background-color: #383838;\n",
       "                }\n",
       "\n",
       "                .geemap-dark .jupyter-button {\n",
       "                    --jp-layout-color3: #383838;\n",
       "                }\n",
       "\n",
       "                .geemap-colab {\n",
       "                    background-color: var(--colab-primary-surface-color, white);\n",
       "                }\n",
       "\n",
       "                .geemap-colab .jupyter-button {\n",
       "                    --jp-layout-color3: var(--colab-primary-surface-color, white);\n",
       "                }\n",
       "            </style>\n",
       "            "
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import ee\n",
    "\n",
    "# Trigger the authentication flow.\n",
    "ee.Authenticate()\n",
    "\n",
    "# Initialize the library.\n",
    "ee.Initialize(project='ee-jvg')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting geemap\n",
      "  Using cached geemap-0.33.1-py2.py3-none-any.whl.metadata (11 kB)\n",
      "Collecting bqplot (from geemap)\n",
      "  Using cached bqplot-0.12.43-py2.py3-none-any.whl.metadata (6.4 kB)\n",
      "Collecting colour (from geemap)\n",
      "  Using cached colour-0.1.5-py2.py3-none-any.whl.metadata (18 kB)\n",
      "Collecting earthengine-api>=0.1.347 (from geemap)\n",
      "  Downloading earthengine_api-0.1.415-py3-none-any.whl.metadata (1.8 kB)\n",
      "Collecting eerepr>=0.0.4 (from geemap)\n",
      "  Using cached eerepr-0.0.4-py3-none-any.whl.metadata (4.0 kB)\n",
      "Collecting folium>=0.13.0 (from geemap)\n",
      "  Using cached folium-0.17.0-py2.py3-none-any.whl.metadata (3.8 kB)\n",
      "Collecting geocoder (from geemap)\n",
      "  Using cached geocoder-1.38.1-py2.py3-none-any.whl.metadata (14 kB)\n",
      "Collecting ipyevents (from geemap)\n",
      "  Using cached ipyevents-2.0.2-py3-none-any.whl.metadata (2.9 kB)\n",
      "Collecting ipyfilechooser>=0.6.0 (from geemap)\n",
      "  Using cached ipyfilechooser-0.6.0-py3-none-any.whl.metadata (6.4 kB)\n",
      "Collecting ipyleaflet>=0.17.0 (from geemap)\n",
      "  Using cached ipyleaflet-0.19.2-py3-none-any.whl.metadata (5.3 kB)\n",
      "Collecting ipytree (from geemap)\n",
      "  Using cached ipytree-0.2.2-py2.py3-none-any.whl.metadata (849 bytes)\n",
      "Requirement already satisfied: matplotlib in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from geemap) (3.6.3)\n",
      "Requirement already satisfied: numpy in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from geemap) (1.24.3)\n",
      "Requirement already satisfied: pandas in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from geemap) (2.0.2)\n",
      "Collecting plotly (from geemap)\n",
      "  Using cached plotly-5.23.0-py3-none-any.whl.metadata (7.3 kB)\n",
      "Collecting pyperclip (from geemap)\n",
      "  Using cached pyperclip-1.9.0.tar.gz (20 kB)\n",
      "  Preparing metadata (setup.py): started\n",
      "  Preparing metadata (setup.py): finished with status 'done'\n",
      "Requirement already satisfied: pyshp>=2.1.3 in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from geemap) (2.3.1)\n",
      "Collecting python-box (from geemap)\n",
      "  Downloading python_box-7.2.0-cp311-cp311-win_amd64.whl.metadata (8.0 kB)\n",
      "Collecting scooby (from geemap)\n",
      "  Using cached scooby-0.10.0-py3-none-any.whl.metadata (15 kB)\n",
      "Collecting google-cloud-storage (from earthengine-api>=0.1.347->geemap)\n",
      "  Using cached google_cloud_storage-2.18.0-py2.py3-none-any.whl.metadata (9.1 kB)\n",
      "Collecting google-api-python-client>=1.12.1 (from earthengine-api>=0.1.347->geemap)\n",
      "  Downloading google_api_python_client-2.140.0-py2.py3-none-any.whl.metadata (6.7 kB)\n",
      "Collecting google-auth>=1.4.1 (from earthengine-api>=0.1.347->geemap)\n",
      "  Using cached google_auth-2.32.0-py2.py3-none-any.whl.metadata (4.7 kB)\n",
      "Collecting google-auth-httplib2>=0.0.3 (from earthengine-api>=0.1.347->geemap)\n",
      "  Using cached google_auth_httplib2-0.2.0-py2.py3-none-any.whl.metadata (2.2 kB)\n",
      "Collecting httplib2<1dev,>=0.9.2 (from earthengine-api>=0.1.347->geemap)\n",
      "  Using cached httplib2-0.22.0-py3-none-any.whl.metadata (2.6 kB)\n",
      "Requirement already satisfied: requests in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from earthengine-api>=0.1.347->geemap) (2.31.0)\n",
      "Collecting branca>=0.6.0 (from folium>=0.13.0->geemap)\n",
      "  Using cached branca-0.7.2-py3-none-any.whl.metadata (1.5 kB)\n",
      "Requirement already satisfied: jinja2>=2.9 in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from folium>=0.13.0->geemap) (3.1.3)\n",
      "Collecting xyzservices (from folium>=0.13.0->geemap)\n",
      "  Using cached xyzservices-2024.6.0-py3-none-any.whl.metadata (4.0 kB)\n",
      "Requirement already satisfied: ipywidgets in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from ipyfilechooser>=0.6.0->geemap) (7.7.2)\n",
      "Collecting jupyter-leaflet<0.20,>=0.19 (from ipyleaflet>=0.17.0->geemap)\n",
      "  Using cached jupyter_leaflet-0.19.2-py3-none-any.whl.metadata (2.4 kB)\n",
      "Collecting traittypes<3,>=0.2.1 (from ipyleaflet>=0.17.0->geemap)\n",
      "  Using cached traittypes-0.2.1-py2.py3-none-any.whl.metadata (1.0 kB)\n",
      "Requirement already satisfied: traitlets>=4.3.0 in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from bqplot->geemap) (5.5.0)\n",
      "Requirement already satisfied: python-dateutil>=2.8.2 in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from pandas->geemap) (2.8.2)\n",
      "Requirement already satisfied: pytz>=2020.1 in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages\\pytz-2022.6-py3.11.egg (from pandas->geemap) (2022.6)\n",
      "Collecting tzdata>=2022.1 (from pandas->geemap)\n",
      "  Downloading tzdata-2024.1-py2.py3-none-any.whl.metadata (1.4 kB)\n",
      "Requirement already satisfied: click in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from geocoder->geemap) (8.1.7)\n",
      "Requirement already satisfied: future in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages\\future-0.18.3-py3.11.egg (from geocoder->geemap) (0.18.3)\n",
      "Collecting ratelim (from geocoder->geemap)\n",
      "  Using cached ratelim-0.1.6-py2.py3-none-any.whl.metadata (1.4 kB)\n",
      "Requirement already satisfied: six in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from geocoder->geemap) (1.16.0)\n",
      "Requirement already satisfied: contourpy>=1.0.1 in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from matplotlib->geemap) (1.2.0)\n",
      "Requirement already satisfied: cycler>=0.10 in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from matplotlib->geemap) (0.11.0)\n",
      "Requirement already satisfied: fonttools>=4.22.0 in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from matplotlib->geemap) (4.25.0)\n",
      "Requirement already satisfied: kiwisolver>=1.0.1 in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from matplotlib->geemap) (1.4.4)\n",
      "Requirement already satisfied: packaging>=20.0 in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from matplotlib->geemap) (23.2)\n",
      "Requirement already satisfied: pillow>=6.2.0 in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from matplotlib->geemap) (10.2.0)\n",
      "Requirement already satisfied: pyparsing>=2.2.1 in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from matplotlib->geemap) (3.0.9)\n",
      "Collecting tenacity>=6.2.0 (from plotly->geemap)\n",
      "  Using cached tenacity-9.0.0-py3-none-any.whl.metadata (1.2 kB)\n",
      "Collecting google-api-core!=2.0.*,!=2.1.*,!=2.2.*,!=2.3.0,<3.0.0.dev0,>=1.31.5 (from google-api-python-client>=1.12.1->earthengine-api>=0.1.347->geemap)\n",
      "  Using cached google_api_core-2.19.1-py3-none-any.whl.metadata (2.7 kB)\n",
      "Collecting uritemplate<5,>=3.0.1 (from google-api-python-client>=1.12.1->earthengine-api>=0.1.347->geemap)\n",
      "  Using cached uritemplate-4.1.1-py2.py3-none-any.whl.metadata (2.9 kB)\n",
      "Requirement already satisfied: cachetools<6.0,>=2.0.0 in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from google-auth>=1.4.1->earthengine-api>=0.1.347->geemap) (4.2.2)\n",
      "Collecting pyasn1-modules>=0.2.1 (from google-auth>=1.4.1->earthengine-api>=0.1.347->geemap)\n",
      "  Using cached pyasn1_modules-0.4.0-py3-none-any.whl.metadata (3.4 kB)\n",
      "Collecting rsa<5,>=3.1.4 (from google-auth>=1.4.1->earthengine-api>=0.1.347->geemap)\n",
      "  Using cached rsa-4.9-py3-none-any.whl.metadata (4.2 kB)\n",
      "Requirement already satisfied: ipykernel>=4.5.1 in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from ipywidgets->ipyfilechooser>=0.6.0->geemap) (6.16.0)\n",
      "Requirement already satisfied: ipython-genutils~=0.2.0 in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from ipywidgets->ipyfilechooser>=0.6.0->geemap) (0.2.0)\n",
      "Requirement already satisfied: widgetsnbextension~=3.6.0 in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from ipywidgets->ipyfilechooser>=0.6.0->geemap) (3.6.1)\n",
      "Requirement already satisfied: ipython>=4.0.0 in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from ipywidgets->ipyfilechooser>=0.6.0->geemap) (7.34.0)\n",
      "Collecting jupyterlab-widgets<3,>=1.0.0 (from ipywidgets->ipyfilechooser>=0.6.0->geemap)\n",
      "  Downloading jupyterlab_widgets-1.1.9-py3-none-any.whl.metadata (3.7 kB)\n",
      "Requirement already satisfied: MarkupSafe>=2.0 in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from jinja2>=2.9->folium>=0.13.0->geemap) (2.1.3)\n",
      "Requirement already satisfied: colorama in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from click->geocoder->geemap) (0.4.6)\n",
      "Collecting google-cloud-core<3.0dev,>=2.3.0 (from google-cloud-storage->earthengine-api>=0.1.347->geemap)\n",
      "  Using cached google_cloud_core-2.4.1-py2.py3-none-any.whl.metadata (2.7 kB)\n",
      "Collecting google-resumable-media>=2.6.0 (from google-cloud-storage->earthengine-api>=0.1.347->geemap)\n",
      "  Using cached google_resumable_media-2.7.1-py2.py3-none-any.whl.metadata (2.2 kB)\n",
      "Collecting google-crc32c<2.0dev,>=1.0 (from google-cloud-storage->earthengine-api>=0.1.347->geemap)\n",
      "  Downloading google_crc32c-1.5.0-cp311-cp311-win_amd64.whl.metadata (2.3 kB)\n",
      "Requirement already satisfied: charset-normalizer<4,>=2 in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from requests->earthengine-api>=0.1.347->geemap) (2.0.4)\n",
      "Requirement already satisfied: idna<4,>=2.5 in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from requests->earthengine-api>=0.1.347->geemap) (3.4)\n",
      "Requirement already satisfied: urllib3<3,>=1.21.1 in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from requests->earthengine-api>=0.1.347->geemap) (2.1.0)\n",
      "Requirement already satisfied: certifi>=2017.4.17 in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from requests->earthengine-api>=0.1.347->geemap) (2024.2.2)\n",
      "Requirement already satisfied: decorator in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from ratelim->geocoder->geemap) (5.1.1)\n",
      "Collecting googleapis-common-protos<2.0.dev0,>=1.56.2 (from google-api-core!=2.0.*,!=2.1.*,!=2.2.*,!=2.3.0,<3.0.0.dev0,>=1.31.5->google-api-python-client>=1.12.1->earthengine-api>=0.1.347->geemap)\n",
      "  Using cached googleapis_common_protos-1.63.2-py2.py3-none-any.whl.metadata (1.5 kB)\n",
      "Requirement already satisfied: protobuf!=3.20.0,!=3.20.1,!=4.21.0,!=4.21.1,!=4.21.2,!=4.21.3,!=4.21.4,!=4.21.5,<6.0.0.dev0,>=3.19.5 in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from google-api-core!=2.0.*,!=2.1.*,!=2.2.*,!=2.3.0,<3.0.0.dev0,>=1.31.5->google-api-python-client>=1.12.1->earthengine-api>=0.1.347->geemap) (4.21.12)\n",
      "Collecting proto-plus<2.0.0dev,>=1.22.3 (from google-api-core!=2.0.*,!=2.1.*,!=2.2.*,!=2.3.0,<3.0.0.dev0,>=1.31.5->google-api-python-client>=1.12.1->earthengine-api>=0.1.347->geemap)\n",
      "  Using cached proto_plus-1.24.0-py3-none-any.whl.metadata (2.2 kB)\n",
      "Requirement already satisfied: debugpy>=1.0 in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from ipykernel>=4.5.1->ipywidgets->ipyfilechooser>=0.6.0->geemap) (1.6.7)\n",
      "Requirement already satisfied: jupyter-client>=6.1.12 in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from ipykernel>=4.5.1->ipywidgets->ipyfilechooser>=0.6.0->geemap) (6.1.12)\n",
      "Requirement already satisfied: matplotlib-inline>=0.1 in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from ipykernel>=4.5.1->ipywidgets->ipyfilechooser>=0.6.0->geemap) (0.1.6)\n",
      "Requirement already satisfied: nest-asyncio in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from ipykernel>=4.5.1->ipywidgets->ipyfilechooser>=0.6.0->geemap) (1.6.0)\n",
      "Requirement already satisfied: psutil in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from ipykernel>=4.5.1->ipywidgets->ipyfilechooser>=0.6.0->geemap) (5.9.0)\n",
      "Requirement already satisfied: pyzmq>=17 in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from ipykernel>=4.5.1->ipywidgets->ipyfilechooser>=0.6.0->geemap) (25.0.2)\n",
      "Requirement already satisfied: tornado>=6.1 in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from ipykernel>=4.5.1->ipywidgets->ipyfilechooser>=0.6.0->geemap) (6.3.3)\n",
      "Requirement already satisfied: setuptools>=18.5 in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from ipython>=4.0.0->ipywidgets->ipyfilechooser>=0.6.0->geemap) (68.2.2)\n",
      "Requirement already satisfied: jedi>=0.16 in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from ipython>=4.0.0->ipywidgets->ipyfilechooser>=0.6.0->geemap) (0.18.2)\n",
      "Requirement already satisfied: pickleshare in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from ipython>=4.0.0->ipywidgets->ipyfilechooser>=0.6.0->geemap) (0.7.5)\n",
      "Requirement already satisfied: prompt-toolkit!=3.0.0,!=3.0.1,<3.1.0,>=2.0.0 in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from ipython>=4.0.0->ipywidgets->ipyfilechooser>=0.6.0->geemap) (3.0.42)\n",
      "Requirement already satisfied: pygments in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from ipython>=4.0.0->ipywidgets->ipyfilechooser>=0.6.0->geemap) (2.17.2)\n",
      "Requirement already satisfied: backcall in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from ipython>=4.0.0->ipywidgets->ipyfilechooser>=0.6.0->geemap) (0.2.0)\n",
      "Collecting pyasn1<0.7.0,>=0.4.6 (from pyasn1-modules>=0.2.1->google-auth>=1.4.1->earthengine-api>=0.1.347->geemap)\n",
      "  Using cached pyasn1-0.6.0-py2.py3-none-any.whl.metadata (8.3 kB)\n",
      "Requirement already satisfied: notebook>=4.4.1 in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from widgetsnbextension~=3.6.0->ipywidgets->ipyfilechooser>=0.6.0->geemap) (6.4.12)\n",
      "Requirement already satisfied: parso<0.9.0,>=0.8.0 in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from jedi>=0.16->ipython>=4.0.0->ipywidgets->ipyfilechooser>=0.6.0->geemap) (0.8.3)\n",
      "Requirement already satisfied: jupyter-core>=4.6.0 in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from jupyter-client>=6.1.12->ipykernel>=4.5.1->ipywidgets->ipyfilechooser>=0.6.0->geemap) (4.11.2)\n",
      "Requirement already satisfied: argon2-cffi in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets->ipyfilechooser>=0.6.0->geemap) (21.3.0)\n",
      "Requirement already satisfied: nbformat in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets->ipyfilechooser>=0.6.0->geemap) (5.6.1)\n",
      "Requirement already satisfied: nbconvert>=5 in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets->ipyfilechooser>=0.6.0->geemap) (7.2.1)\n",
      "Requirement already satisfied: Send2Trash>=1.8.0 in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets->ipyfilechooser>=0.6.0->geemap) (1.8.2)\n",
      "Requirement already satisfied: terminado>=0.8.3 in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets->ipyfilechooser>=0.6.0->geemap) (0.17.1)\n",
      "Requirement already satisfied: prometheus-client in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets->ipyfilechooser>=0.6.0->geemap) (0.8.0)\n",
      "Requirement already satisfied: wcwidth in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from prompt-toolkit!=3.0.0,!=3.0.1,<3.1.0,>=2.0.0->ipython>=4.0.0->ipywidgets->ipyfilechooser>=0.6.0->geemap) (0.2.5)\n",
      "Requirement already satisfied: beautifulsoup4 in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from nbconvert>=5->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets->ipyfilechooser>=0.6.0->geemap) (4.12.2)\n",
      "Requirement already satisfied: bleach in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from nbconvert>=5->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets->ipyfilechooser>=0.6.0->geemap) (4.1.0)\n",
      "Requirement already satisfied: defusedxml in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from nbconvert>=5->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets->ipyfilechooser>=0.6.0->geemap) (0.7.1)\n",
      "Requirement already satisfied: jupyterlab-pygments in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from nbconvert>=5->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets->ipyfilechooser>=0.6.0->geemap) (0.2.2)\n",
      "Requirement already satisfied: mistune<3,>=2.0.3 in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from nbconvert>=5->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets->ipyfilechooser>=0.6.0->geemap) (2.0.4)\n",
      "Requirement already satisfied: nbclient>=0.5.0 in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from nbconvert>=5->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets->ipyfilechooser>=0.6.0->geemap) (0.7.0)\n",
      "Requirement already satisfied: pandocfilters>=1.4.1 in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from nbconvert>=5->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets->ipyfilechooser>=0.6.0->geemap) (1.5.0)\n",
      "Requirement already satisfied: tinycss2 in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from nbconvert>=5->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets->ipyfilechooser>=0.6.0->geemap) (1.1.1)\n",
      "Requirement already satisfied: fastjsonschema in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from nbformat->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets->ipyfilechooser>=0.6.0->geemap) (2.16.2)\n",
      "Requirement already satisfied: jsonschema>=2.6 in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from nbformat->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets->ipyfilechooser>=0.6.0->geemap) (4.19.2)\n",
      "Requirement already satisfied: pywinpty>=1.1.0 in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from terminado>=0.8.3->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets->ipyfilechooser>=0.6.0->geemap) (2.0.10)\n",
      "Requirement already satisfied: argon2-cffi-bindings in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from argon2-cffi->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets->ipyfilechooser>=0.6.0->geemap) (21.2.0)\n",
      "Requirement already satisfied: attrs>=22.2.0 in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from jsonschema>=2.6->nbformat->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets->ipyfilechooser>=0.6.0->geemap) (23.1.0)\n",
      "Requirement already satisfied: jsonschema-specifications>=2023.03.6 in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from jsonschema>=2.6->nbformat->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets->ipyfilechooser>=0.6.0->geemap) (2023.7.1)\n",
      "Requirement already satisfied: referencing>=0.28.4 in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from jsonschema>=2.6->nbformat->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets->ipyfilechooser>=0.6.0->geemap) (0.30.2)\n",
      "Requirement already satisfied: rpds-py>=0.7.1 in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from jsonschema>=2.6->nbformat->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets->ipyfilechooser>=0.6.0->geemap) (0.10.6)\n",
      "Requirement already satisfied: cffi>=1.0.1 in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from argon2-cffi-bindings->argon2-cffi->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets->ipyfilechooser>=0.6.0->geemap) (1.16.0)\n",
      "Requirement already satisfied: soupsieve>1.2 in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from beautifulsoup4->nbconvert>=5->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets->ipyfilechooser>=0.6.0->geemap) (2.5)\n",
      "Requirement already satisfied: webencodings in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from bleach->nbconvert>=5->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets->ipyfilechooser>=0.6.0->geemap) (0.5.1)\n",
      "Requirement already satisfied: pycparser in c:\\users\\jvalenciag\\appdata\\local\\esri\\conda\\envs\\arcgispro-py3-clone\\lib\\site-packages (from cffi>=1.0.1->argon2-cffi-bindings->argon2-cffi->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets->ipyfilechooser>=0.6.0->geemap) (2.21)\n",
      "Using cached geemap-0.33.1-py2.py3-none-any.whl (2.2 MB)\n",
      "Downloading earthengine_api-0.1.415-py3-none-any.whl (450 kB)\n",
      "   ---------------------------------------- 450.2/450.2 kB 1.2 MB/s eta 0:00:00\n",
      "Using cached eerepr-0.0.4-py3-none-any.whl (9.7 kB)\n",
      "Using cached folium-0.17.0-py2.py3-none-any.whl (108 kB)\n",
      "Using cached ipyfilechooser-0.6.0-py3-none-any.whl (11 kB)\n",
      "Using cached ipyleaflet-0.19.2-py3-none-any.whl (31 kB)\n",
      "Using cached bqplot-0.12.43-py2.py3-none-any.whl (1.2 MB)\n",
      "Using cached colour-0.1.5-py2.py3-none-any.whl (23 kB)\n",
      "Using cached geocoder-1.38.1-py2.py3-none-any.whl (98 kB)\n",
      "Using cached ipyevents-2.0.2-py3-none-any.whl (101 kB)\n",
      "Using cached ipytree-0.2.2-py2.py3-none-any.whl (1.3 MB)\n",
      "Using cached plotly-5.23.0-py3-none-any.whl (17.3 MB)\n",
      "Downloading python_box-7.2.0-cp311-cp311-win_amd64.whl (1.2 MB)\n",
      "   ---------------------------------------- 1.2/1.2 MB 4.0 MB/s eta 0:00:00\n",
      "Using cached scooby-0.10.0-py3-none-any.whl (18 kB)\n",
      "Using cached branca-0.7.2-py3-none-any.whl (25 kB)\n",
      "Downloading google_api_python_client-2.140.0-py2.py3-none-any.whl (12.1 MB)\n",
      "   ---------------------------------------- 12.1/12.1 MB 13.9 MB/s eta 0:00:00\n",
      "Using cached google_auth-2.32.0-py2.py3-none-any.whl (195 kB)\n",
      "Using cached google_auth_httplib2-0.2.0-py2.py3-none-any.whl (9.3 kB)\n",
      "Using cached httplib2-0.22.0-py3-none-any.whl (96 kB)\n",
      "Using cached jupyter_leaflet-0.19.2-py3-none-any.whl (1.1 MB)\n",
      "Using cached tenacity-9.0.0-py3-none-any.whl (28 kB)\n",
      "Using cached traittypes-0.2.1-py2.py3-none-any.whl (8.6 kB)\n",
      "Downloading tzdata-2024.1-py2.py3-none-any.whl (345 kB)\n",
      "   ---------------------------------------- 345.4/345.4 kB 7.1 MB/s eta 0:00:00\n",
      "Using cached xyzservices-2024.6.0-py3-none-any.whl (83 kB)\n",
      "Using cached google_cloud_storage-2.18.0-py2.py3-none-any.whl (130 kB)\n",
      "Using cached ratelim-0.1.6-py2.py3-none-any.whl (4.0 kB)\n",
      "Using cached google_api_core-2.19.1-py3-none-any.whl (139 kB)\n",
      "Using cached google_cloud_core-2.4.1-py2.py3-none-any.whl (29 kB)\n",
      "Downloading google_crc32c-1.5.0-cp311-cp311-win_amd64.whl (27 kB)\n",
      "Using cached google_resumable_media-2.7.1-py2.py3-none-any.whl (81 kB)\n",
      "Downloading jupyterlab_widgets-1.1.9-py3-none-any.whl (237 kB)\n",
      "   ---------------------------------------- 238.0/238.0 kB 3.7 MB/s eta 0:00:00\n",
      "Using cached pyasn1_modules-0.4.0-py3-none-any.whl (181 kB)\n",
      "Using cached rsa-4.9-py3-none-any.whl (34 kB)\n",
      "Using cached uritemplate-4.1.1-py2.py3-none-any.whl (10 kB)\n",
      "Using cached googleapis_common_protos-1.63.2-py2.py3-none-any.whl (220 kB)\n",
      "Using cached proto_plus-1.24.0-py3-none-any.whl (50 kB)\n",
      "Using cached pyasn1-0.6.0-py2.py3-none-any.whl (85 kB)\n",
      "Building wheels for collected packages: pyperclip\n",
      "  Building wheel for pyperclip (setup.py): started\n",
      "  Building wheel for pyperclip (setup.py): finished with status 'done'\n",
      "  Created wheel for pyperclip: filename=pyperclip-1.9.0-py3-none-any.whl size=11019 sha256=1ba1da1e2e44b2116808c961c86b91497ff6036968d5dfa989d1827859c847f0\n",
      "  Stored in directory: c:\\users\\jvalenciag\\appdata\\local\\pip\\cache\\wheels\\e8\\e7\\56\\591cb88ba1783b38c40d584026e766aac9c3a048e34128ce8b\n",
      "Successfully built pyperclip\n",
      "Installing collected packages: pyperclip, colour, xyzservices, uritemplate, tzdata, traittypes, tenacity, scooby, ratelim, python-box, pyasn1, proto-plus, jupyterlab-widgets, jupyter-leaflet, httplib2, googleapis-common-protos, google-crc32c, rsa, pyasn1-modules, plotly, google-resumable-media, geocoder, branca, google-auth, folium, google-auth-httplib2, google-api-core, google-cloud-core, google-api-python-client, google-cloud-storage, earthengine-api, eerepr, ipytree, ipyleaflet, ipyfilechooser, ipyevents, bqplot, geemap\n",
      "  Attempting uninstall: jupyterlab-widgets\n",
      "    Found existing installation: jupyterlab-widgets 3.0.3\n",
      "    Uninstalling jupyterlab-widgets-3.0.3:\n",
      "      Successfully uninstalled jupyterlab-widgets-3.0.3\n",
      "Successfully installed bqplot-0.12.43 branca-0.7.2 colour-0.1.5 earthengine-api-0.1.415 eerepr-0.0.4 folium-0.17.0 geemap-0.33.1 geocoder-1.38.1 google-api-core-2.19.1 google-api-python-client-2.140.0 google-auth-2.32.0 google-auth-httplib2-0.2.0 google-cloud-core-2.4.1 google-cloud-storage-2.18.0 google-crc32c-1.5.0 google-resumable-media-2.7.1 googleapis-common-protos-1.63.2 httplib2-0.22.0 ipyevents-2.0.2 ipyfilechooser-0.6.0 ipyleaflet-0.19.2 ipytree-0.2.2 jupyter-leaflet-0.19.2 jupyterlab-widgets-1.1.9 plotly-5.23.0 proto-plus-1.24.0 pyasn1-0.6.0 pyasn1-modules-0.4.0 pyperclip-1.9.0 python-box-7.2.0 ratelim-0.1.6 rsa-4.9 scooby-0.10.0 tenacity-9.0.0 traittypes-0.2.1 tzdata-2024.1 uritemplate-4.1.1 xyzservices-2024.6.0\n",
      "Name: geemap\n",
      "Version: 0.33.1\n",
      "Summary: A Python package for interactive mapping using Google Earth Engine and ipyleaflet\n",
      "Home-page: \n",
      "Author: \n",
      "Author-email: Qiusheng Wu <giswqs@gmail.com>\n",
      "License: MIT\n",
      "Location: C:\\Users\\jvalenciag\\AppData\\Local\\ESRI\\conda\\envs\\arcgispro-py3-clone\\Lib\\site-packages\n",
      "Requires: bqplot, colour, earthengine-api, eerepr, folium, geocoder, ipyevents, ipyfilechooser, ipyleaflet, ipytree, matplotlib, numpy, pandas, plotly, pyperclip, pyshp, python-box, scooby\n",
      "Required-by: \n"
     ]
    }
   ],
   "source": [
    "# If not installed, run this cell. It helps visualization of data\n",
    "!pip install geemap\n",
    "!pip show geemap"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "d6f4c18d10bc441f980ced8c3abe9933",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Map(center=[0, 0], controls=(WidgetControl(options=['position', 'transparent_bg'], widget=SearchDataGUI(childr…"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import geemap\n",
    "Map = geemap.Map()\n",
    "Map"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "            <style>\n",
       "                .geemap-dark {\n",
       "                    --jp-widgets-color: white;\n",
       "                    --jp-widgets-label-color: white;\n",
       "                    --jp-ui-font-color1: white;\n",
       "                    --jp-layout-color2: #454545;\n",
       "                    background-color: #383838;\n",
       "                }\n",
       "\n",
       "                .geemap-dark .jupyter-button {\n",
       "                    --jp-layout-color3: #383838;\n",
       "                }\n",
       "\n",
       "                .geemap-colab {\n",
       "                    background-color: var(--colab-primary-surface-color, white);\n",
       "                }\n",
       "\n",
       "                .geemap-colab .jupyter-button {\n",
       "                    --jp-layout-color3: var(--colab-primary-surface-color, white);\n",
       "                }\n",
       "            </style>\n",
       "            "
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Access an image asset from my account\n",
    "guate_bdry = ee.FeatureCollection('users/jvg/guatemala_buffer')\n",
    "guate_geom = guate_bdry.geometry();\n",
    "Map.centerObject(guate_geom);\n",
    "Map.addLayer(guate_bdry, {\"color\": 'red'}, 'Guatemala Boundary');"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "            <style>\n",
       "                .geemap-dark {\n",
       "                    --jp-widgets-color: white;\n",
       "                    --jp-widgets-label-color: white;\n",
       "                    --jp-ui-font-color1: white;\n",
       "                    --jp-layout-color2: #454545;\n",
       "                    background-color: #383838;\n",
       "                }\n",
       "\n",
       "                .geemap-dark .jupyter-button {\n",
       "                    --jp-layout-color3: #383838;\n",
       "                }\n",
       "\n",
       "                .geemap-colab {\n",
       "                    background-color: var(--colab-primary-surface-color, white);\n",
       "                }\n",
       "\n",
       "                .geemap-colab .jupyter-button {\n",
       "                    --jp-layout-color3: var(--colab-primary-surface-color, white);\n",
       "                }\n",
       "            </style>\n",
       "            "
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "clay = ee.Image('OpenLandMap/SOL/SOL_CLAY-WFRACTION_USDA-3A1A1A_M/v02')\n",
    "clay_clipped = clay.clip(guate_bdry)\n",
    "Map.addLayer(clay_clipped, {}, 'clay_clipped')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Let's import other libraries\n",
    "import pandas as pd"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
